src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -143,4 +153,35 @@ void OutgoingMigration::SyncFb() {
   // TODO add ACK here and config update
 }
 
+std::error_code OutgoingMigration::Start(ConnectionContext* cntx) {
+  VLOG(1) << "Starting outgoing migration";
+
+  auto check_connection_error = [&cntx](error_code ec, const char* msg) -> error_code {
+    if (ec) {
+      cntx->SendError(absl::StrCat(msg, ec.message()));
+    }
+    return ec;
+  };
+
+  VLOG(1) << "Resolving host DNS";
+  error_code ec = ResolveHostDns();
+  RETURN_ON_ERR(check_connection_error(ec, "could not resolve host dns"));
+
+  VLOG(1) << "Connecting to source";
+  ec = ConnectAndAuth(absl::GetFlag(FLAGS_source_connect_timeout_ms) * 1ms, &cntx_);
+  RETURN_ON_ERR(check_connection_error(ec, "couldn't connect to source"));
+
+  VLOG(1) << "Migration initiating";
+  ResetParser(false);
+  auto port = absl::GetFlag(FLAGS_admin_port);
```

Comment:
once the redesign is done will we still need to send the port?
all the connections will be from source to target right?

Reply:
if we do all connections from the source node to target we don't need to send port, but I am still not sure should we send ack from from source to target or vice versa)